src/frontend/fields/list/foldable-list.vue

```diff
@@ -1,68 +1,80 @@
+          class="z-[1] inline-flex items-center rounded-full border border-gray-300 bg-white px-4 py-1.5 text-sm font-medium leading-5 text-gray-700 shadow-sm hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-indigo-500"
+          :disabled="isReadOnly"
+          :class="{ 'opacity-0': isReadOnly }"
           @click.prevent="toggle(index)"
         >
           <icon
             v-if="isExpanded(index)"
             name="chevron-down"
-            class="mr-1 icon"
+            class="icon mr-1"
             aria-hidden="true"
           />
-          <icon v-else name="chevron-right" class="mr-1 icon" aria-hidden="true" />
+          <icon v-else name="chevron-right" class="icon mr-1" aria-hidden="true" />
           <span>
             {{ String(sectionTitle(index)) }}
           </span>
         </button>
-        <div v-if="itemHasError(index)" class="text-accent-one">
-          <div class="p-2 bg-white border rounded-full">
-            <Icon name="exclamation" class="w-10 h-10 text-red-500" />
+        <div v-if="itemHasError(index)" class="z-[1] text-accent-one">
+          <div class="rounded-full border bg-white p-2">
+            <Icon name="exclamation" class="h-10 w-10 text-red-500" />
           </div>
         </div>
         <button
           v-if="canMutate"
           type="button"
-          class="flex items-center justify-center w-10 h-10 text-gray-500 bg-white border rounded-full cursor-pointer"
+          class="z-[1] flex h-10 w-10 cursor-pointer items-center justify-center rounded-full border bg-white text-gray-500"
           @click="emit('removeSet', index)"
         >
-          <span v-if="!isReadOnly" class="flex items-center justify-center w-10 h-10">
-            <Icon name="trash" class="w-auto h-auto" />
+          <span v-if="!isReadOnly" class="flex h-10 w-10 items-center justify-center">
+            <Icon name="trash" class="h-auto w-auto" />
           </span>
         </button>
       </div>
-      <ul v-if="isExpanded(index)">
-        <li v-for="(item, i) in fields" :key="item.name + `${i.toString()}`" class="grid">
+      <ul
+        v-if="isExpanded(index)"
+        :class="[
+          'row-[span_100] grid grid-rows-subgrid bg-white',
+          { 'ml-8': !isReadOnly },
+        ]"
+      >
+        <li
+          v-for="(item, i) in fields"
+          :key="item.name + `${i.toString()}`"
+          class="grid grid-rows-subgrid"
+          :style="{ gridRow: `span ${listItems.length}` }"
+        >
           <component
             :is="widgets.picker(item.widget)"
             :class="{
-              'rounded border border-gray-200 bg-white drop-shadow-sm':
-                item.widget != 'list',
-              'mt-8 rounded border border-gray-200 bg-white p-8 shadow': isIsland(
-                item.widget,
-              ),
+              'rounded bg-white': item.widget != 'list',
```

Comment:
we must either remove the class attribute here or wrap the `foldable-list` and `flat-list` in a root element